internal/graph/check.go

```diff
@@ -974,23 +974,68 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 }
 
 // checkComputedUserset evaluates the Check request with the rewritten relation (e.g. the computed userset relation).
-func (c *LocalChecker) checkComputedUserset(_ context.Context, req *ResolveCheckRequest, rewrite *openfgav1.Userset_ComputedUserset) CheckHandlerFunc {
-	return func(ctx context.Context) (*ResolveCheckResponse, error) {
-		ctx, span := tracer.Start(ctx, "checkComputedUserset")
-		defer span.End()
+func (c *LocalChecker) checkComputedUserset(ctx context.Context, req *ResolveCheckRequest, rewrite *openfgav1.Userset) CheckHandlerFunc {
+	_, span := tracer.Start(ctx, "checkComputedUserset")
+	defer span.End()
 
-		if ctx.Err() != nil {
-			return nil, ctx.Err()
-		}
+	seen := map[string]struct{}{}
+	typesys, _ := typesystem.TypesystemFromContext(ctx)
+
+	tk := tuple.NewTupleKey(
+		req.GetTupleKey().GetObject(),
+		req.GetTupleKey().GetRelation(),
+		req.GetTupleKey().GetUser(),
+	)
 
-		rewrittenTupleKey := tuple.NewTupleKey(
-			req.GetTupleKey().GetObject(),
-			rewrite.ComputedUserset.GetRelation(),
-			req.GetTupleKey().GetUser(),
+	for {
+		tk = tuple.NewTupleKey(
+			tk.GetObject(),
+			rewrite.GetComputedUserset().GetRelation(),
+			tk.GetUser(),
 		)
+		key := tuple.TupleKeyToString(tk)
+		if _, cycleDetected := seen[key]; cycleDetected {
```

Comment:
i'm OK with duplicating cycle detection code here, but what do you think of sending a follow-up PR where we merge the cycle resolver in this local resolver and get rid of this?

Reply:
Track via https://github.com/openfga/openfga/issues/1794